lib/types/channels.d.ts

```diff
@@ -476,24 +489,67 @@ export interface URLButton extends ButtonBase {
     url: string;
 }
 
-export interface RawSelectMenu {
+export interface RawSelectMenuBase {
     custom_id: string;
     disabled?: boolean;
     max_values?: number;
     min_values?: number;
-    options: Array<SelectOption>;
     placeholder?: string;
-    type: ComponentTypes.SELECT_MENU;
+    type: SelectMenuTypes;
+}
+
+export interface RawStringSelectMenu extends RawSelectMenuBase {
+    options: Array<SelectOption>;
+    type: ComponentTypes.STRING_SELECT;
 }
 
-export interface SelectMenu {
+export interface RawUserSelectMenu extends RawSelectMenuBase {
+    type: ComponentTypes.USER_SELECT;
+}
+
+export interface RawRoleSelectMenu extends RawSelectMenuBase {
+    type: ComponentTypes.ROLE_SELECT;
+}
+
+export interface RawMentionableSelectMenu extends RawSelectMenuBase {
+    type: ComponentTypes.MENTIONABLE_SELECT;
+}
```

Comment:
Make `RawSelectMenuBase` have a generic parameter which is the type, this eliminates the needs for the mess of extends and turns these into one liners